From a3f491c2f1a9b8a917f960a64712a5f7a769eff0 Mon Sep 17 00:00:00 2001
From: Sandeep Patil <sspatil@google.com>
Date: Mon, 23 Mar 2020 20:40:00 -0700
Subject: [PATCH 670/830] ANDROID: GKI: Add CONFIG_SOC_BUS to gki_defconfig
X-NVConfidentiality: public

One of the hidden defconfigs that must be enabled for SoC
devices to be able to register themselves with the SoC bus.

Bug: 152265482
Test: Builds

Signed-off-by: Sandeep Patil <sspatil@google.com>
Change-Id: Ia56f4b0618f8d64acdf44cbb346313119460f745
(cherry picked from commit 1f3f85de1c59b1cc03b0e0067855c367c97a7cc5)
---
 init/Kconfig.gki | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/init/Kconfig.gki b/init/Kconfig.gki
index c752d6284773..1a43c5da3309 100644
--- a/init/Kconfig.gki
+++ b/init/Kconfig.gki
@@ -127,6 +127,16 @@ config GKI_HIDDEN_USB_CONFIGS
 	  If you are not building a kernel to be used for a variety of
 	  out-of-kernel build USB drivers, say N here.
 
+config GKI_HIDDEN_SOC_BUS_CONFIGS
+	bool "Hidden SoC bus configuration needed for GKI"
+	select SOC_BUS
+	  help
+	    Dummy config option used to enable SOC_BUS hidden Kconfig.
+	    The configuration is required for SoCs to register themselves to the bus.
+
+	    If you are not building a kernel to be used for a variety of SoCs and
+	    out-of-tree drivers, say N here.
+
 # Atrocities needed for
 # a) building GKI modules in separate tree, or
 # b) building drivers that are not modularizable
@@ -148,6 +158,7 @@ config GKI_HACKS_TO_FIX
 	select GKI_HIDDEN_MEDIA_CONFIGS
 	select GKI_HIDDEN_VIRTUAL_CONFIGS
 	select GKI_HIDDEN_USB_CONFIGS
+	select GKI_HIDDEN_SOC_BUS_CONFIGS
 	help
 	  Dummy config option used to enable core functionality used by
 	  modules that may not be selectable in this config.
-- 
2.17.1

